athenahealth/appointment_status.go

```diff
@@ -0,0 +1,37 @@
+package athenahealth
+
+// AppointmentStatus is derived from https://docs.athenahealth.com/api/api-ref/appointment#Book-appointment
+// The athenaNet appointment status. There are several possible statuses.
+// x=cancelled
+// f=future (It can include appointments where were never checked in, even if the appointment date is in the past. It is up to a practice to cancel appointments as a no show when appropriate to do so.)
+// o=open
+// 2=checked in
+// 3=checked out
+// 4=charge entered (i.e. a past appointment)
+type AppointmentStatus string
+
+const (
+	AppointmentStatusCancelled     AppointmentStatus = "x"
+	AppointmentStatusChargeEntered AppointmentStatus = "4"
+	AppointmentStatusCheckedIn     AppointmentStatus = "2"
+	AppointmentStatusCheckedOut    AppointmentStatus = "3"
+	AppointmentStatusFuture        AppointmentStatus = "f"
+	AppointmentStatusOpen          AppointmentStatus = "o"
+)
+
+var appointmentStatuses = map[AppointmentStatus]bool{
+	AppointmentStatusCancelled:     true,
+	AppointmentStatusChargeEntered: true,
+	AppointmentStatusCheckedIn:     true,
+	AppointmentStatusCheckedOut:    true,
+	AppointmentStatusFuture:        true,
+	AppointmentStatusOpen:          true,
+}
```

Comment:
Curious - why `bool` instead of an empty struct `struct{}`? `false` would be a valid value but shouldn't dictate any meaningful behavior.
Empty struct is zero allocation and seeing that we don't care about the value it would be more appropriate.
https://go.dev/play/p/s4yNIUEv6G_M
https://dave.cheney.net/2014/03/25/the-empty-struct

```suggestion
var appointmentStatuses = map[AppointmentStatus]struct{}{
	AppointmentStatusCancelled:     {},
	AppointmentStatusChargeEntered: {},
	AppointmentStatusCheckedIn:     {},
	AppointmentStatusCheckedOut:    {},
	AppointmentStatusFuture:        {},
	AppointmentStatusOpen:          {},
}
```

Reply:
mostly because I had no idea this was a thing - this is awesome! thanks I was trying to replace a 'set' from java/typescript.